NEWS.md

```diff
@@ -131,86 +129,87 @@ or possible cast.
+  assertions, and of class `"vctrs_error_incompatible"` (with
+  subclasses `_type`, `_cast` and `_op`) for errors on incompatible
+  types (#184).
+
+* Character indexing is now only supported for named objects, an error
+  is raised for unnamed objects (#171).
+
 * Predicate generics now consistently return logical vectors when
   passed a `vctrs_vctr` class. They used to restore the output to
   their input type (#251).
 
+* `list_of()` now has an `as.character()` method. It uses
+  `vec_ptype_abbr()` to collapse complex objects into their type
+  representation (tidyverse/tidyr#654).
+
 * New `stop_incompatible_size()` to signal a failure due to mismatched sizes.
 
-* `vec_slice<-` now has a `vec_assign()` alias. Use `vec_assign()`
-  when you don't want to modify the original input.
+* New `validate_list_of()` (#193).
 
-* `vec_slice()` now calls `vec_restore()` automatically. Unlike the
-  default `[` method from base R, attributes are preserved by default.
+* `vec_arith()` is consistent with base R when combining `difftime`
+  and `date`, with a warning if casts are lossy (#192).
 
-* `vec_size()` now works with positive short row names. This fixes
-  issues with data frames created with jsonlite (#220).
+* `vec_c()` and `vec_rbind()` now handle data.frame columns properly
+  (@yutannihilation, #182).
 
-* New `validate_list_of()` (#193).
+* `vec_cast(x, data.frame())` preserves the number of rows in `x`.
 
 * `vec_equal()` now handles missing values symmetrically (#204).
 
+* `vec_equal_na()` now returns `TRUE` for data frames and records when
+  every component is missing, not when _any_ component is missing
+  (#201).
+
+* `vec_init()` checks input is a vector.
+
 * `vec_proxy_compare()` gains an experimental `relax` argument, which
   allows data frames to be orderable even if all their columns are not
   (#210).
 
-* New `vec_order()` and `vec_sort()` for ordering and sorting
-  generalised vectors.
+* `vec_size()` now works with positive short row names. This fixes
+  issues with data frames created with jsonlite (#220).
 
-* New `vec_repeat()` for repeating each element of a vector the same number
-  of times.
+* `vec_slice<-` now has a `vec_assign()` alias. Use `vec_assign()`
+  when you don't want to modify the original input.
 
-* `vec_slice()` can correct slice 0-row data frames (#179).
+* `vec_slice()` now calls `vec_restore()` automatically. Unlike the
+  default `[` method from base R, attributes are preserved by default.
 
-* New `vec_seq_along()` and `vec_init_along()` create useful sequences (#189).
+* `vec_slice()` can correct slice 0-row data frames (#179).
 
-* `vec_equal_na()` now returns `TRUE` for data frames and records when
-  every component is mising, not when _any_ component is missing
-  (#201).
+* New `vec_repeat()` for repeating each element of a vector the same number
+  of times.
 
 * `vec_type2(x, data.frame())` ensures that the returned object has
```

Comment:
I might've just missed it, but do you describe the improvements to error messages somewhere?

Reply:
You're thinking about error messages when nested data frames are not congruent right?